plugp100/discovery/tapo_discovery.py

```diff
@@ -0,0 +1,208 @@
+import ipaddress
+import logging
+import struct
+import sys
+import zlib
+import json
+import base64
+import socket
+import io
+import select
+import base64
+import os
+import time
+
+from Crypto.Cipher import PKCS1_v1_5
+from Crypto.Cipher import PKCS1_OAEP
+from Crypto.PublicKey import RSA
+from Crypto.Cipher import AES
+
+from typing import Optional, Generator
+
+logger = logging.getLogger(__name__)
+
+PKT_ONBOARD_REQUEST  = b'\x11\x00'
+PKT_ONBOARD_RESPONSE = b'"\x01'
+
+# the whole discovery protocol is pure obscurity, so we won't bother wasting cpu cycles on generating random RSA keys
+OUR_KEY = RSA.import_key("""-----BEGIN PRIVATE KEY-----
```

Comment:
Firstly, really thanks for your contribution. 

Regarding RSA generation, I think it is better to generate a new RSA key pair, like the actual protocol. Using the same key could lead into future issues by users